File: leetcode_problems/typescript/src/lc1822_sign_of_product_of_any_array.ts
function arraySign(nums: number[]): number {
  let m = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] == 0) {
      return 0;
    } else if (nums[i] < 0) {
      m++;
    }
  }
  if (m % 2 == 0) {
    return 1;
  } else {
    return -1;
  }
}

File: leetcode_problems/cpp/lc1822_sign_of_product_of_any_array.cpp
#include <vector>
#include <iostream>
using std::vector;

class Solution
{
public:
    int arraySign(vector<int> &nums)
    {
        int m = 0;
        for (int i = 0; i < nums.size(); i++)
        {
            if (nums[i] == 0)
            {
                return 0;
            }
            else if (nums[i] < 0)
            {
                m++;
            }
        }
        if (m % 2 == 0)
        {
            return 1;
        }
        else
        {
            return -1;
        }
    }
};

int main()
{
    vector<int> nums = {-1, -2, -3, -4, 3, 2, 1};
    Solution sol;
    std::cout << sol.arraySign(nums) << std::endl;
    return 0;
}